.github/workflows/release.yaml

```diff
@@ -0,0 +1,44 @@
+name: NATS Server Releases
+on:
+  push:
+    tags:
+      - v*
+
+jobs:
+  run:
+    name: GitHub Release
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          fetch-tags: true
+
+      - name: Set up Go
+        uses: actions/setup-go@v5
+        with:
+          go-version: "stable"
+
+      - name: Check version matches tag
+        env:
+          TRAVIS_TAG: ${{ github.ref_name }}
+        run: |
+          go test -race -v -run=TestVersionMatchesTag ./server -ldflags="-X=github.com/nats-io/nats-server/v2/server.serverVersion=$TRAVIS_TAG" -count=1 -vet=off
+
+      - name: Install cosign
+        uses: sigstore/cosign-installer@v3.7.0
+
+      - name: Install syft
+        uses: anchore/sbom-action/download-syft@v0.18.0
+        with:
+          syft-version: "v1.20.0"
+
+      - name: Create release
+        uses: goreleaser/goreleaser-action@v6
```

Comment:
Do you know if the goreleaser behind this will be "pinned" in that the version does not change? If it does, is there a way to pin to be sure we don't get a bad surprise during a release, or alternatively and like I was doing when I was doing releases, I would run the goreleaser locally (with the `--skip-publish` option) to make sure that everything is ok.